jdaviz/tests/test_subsets.py

```diff
@@ -791,6 +792,94 @@ def test_only_overlapping_in_specviz2d(specviz2d_helper, mos_spectrum2d):
     assert reg[1].lower.value == 7600 and reg[1].upper.value == 7800
 
 
+def test_draw2d_linking_specviz2d(specviz2d_helper):
+    # custom test data to predict values for different viewers
+    header = {
+              'WCSAXES': 2,
+              'CRPIX1': 0.0, 'CRPIX2': 8.5,
+              'CDELT1': 1E-06, 'CDELT2': 7.5E-05,
+              'CUNIT1': 'm', 'CUNIT2': 'deg',
+              'CTYPE1': 'WAVE', 'CTYPE2': 'OFFSET',
+              'CRVAL1': 0.0, 'CRVAL2': 5.0,
+              'RADESYS': 'ICRS', 'SPECSYS': 'BARYCENT'}
+    wcs = WCS(header)
+
+    x_values = np.linspace(0, 10, 128)
+    y_values = np.linspace(0, 5, 256)
+
+    # Create a continuous 2D
+    data = np.sin(x_values[:, np.newaxis]) * np.cos(y_values) * u.one
+    spectrum_data = Spectrum1D(data, wcs=wcs, meta=header)
+
+    specviz2d_helper.load_data(spectrum_2d=spectrum_data)
+    viewer_1d = specviz2d_helper.app.get_viewer(
+        specviz2d_helper._default_spectrum_viewer_reference_name)
+    viewer_2d = specviz2d_helper.app.get_viewer(
+        specviz2d_helper._default_spectrum_2d_viewer_reference_name)
+
+    # create subset in 2d viewer,  want data in 1d viewer
+    viewer_2d.apply_roi(XRangeROI(60, 80))
+    subset_drawn_2d = viewer_1d.native_marks[-1]
+
+    # get x and y components to compute subset mask
+    y1 = subset_drawn_2d.y
+    x1 = subset_drawn_2d.x
+
+    subset_highlighted_region1 = x1[np.isfinite(y1)]
+    min_value_subset1 = np.min(subset_highlighted_region1)
+    max_value_subset1 = np.max(subset_highlighted_region1)
+
+    tolerance1 = 1e-6
+    expected_min1 = 6.e-5
+    expected_max1 = 8.e-05
+
+    assert np.allclose(min_value_subset1, expected_min1, atol=tolerance1)
+    assert np.allclose(max_value_subset1, expected_max1, atol=tolerance1)
+
+
+def test_draw1d_linking_specviz2d(specviz2d_helper):
+    # custom test data to predict values for different viewers
+    header = {
+        'WCSAXES': 2,
+        'CRPIX1': 0.0, 'CRPIX2': 8.5,
+        'CDELT1': 1E-06, 'CDELT2': 7.5E-05,
+        'CUNIT1': 'm', 'CUNIT2': 'deg',
+        'CTYPE1': 'WAVE', 'CTYPE2': 'OFFSET',
+        'CRVAL1': 0.0, 'CRVAL2': 5.0,
+        'RADESYS': 'ICRS', 'SPECSYS': 'BARYCENT'}
+    wcs = WCS(header)
+
+    x_values = np.linspace(0, 10, 128)
+    y_values = np.linspace(0, 5, 256)
+
+    # Create a continuous 2D image
+    data = np.sin(x_values[:, np.newaxis]) * np.cos(y_values) * u.one
+    spectrum_data = Spectrum1D(data, wcs=wcs, meta=header)
+
+    specviz2d_helper.load_data(spectrum_2d=spectrum_data)
+    viewer_1d = specviz2d_helper.app.get_viewer(
+        specviz2d_helper._default_spectrum_viewer_reference_name)
+    viewer_2d = specviz2d_helper.app.get_viewer(
+        specviz2d_helper._default_spectrum_2d_viewer_reference_name)
+
+    # subset drawn in 1d viewer, want data in 2d viewer
+    viewer_1d.apply_roi(XRangeROI(.0001, .0002))
+    subset_drawn_1d = viewer_2d.native_marks[-1].image
```

Comment:
Getting the 3d image array this way is loading incorrectly only in the test. It should be (155, 864, 4) [which is how it loads in a Notebook], in the test file it is loading a zero array of shape (10,10,4), so the test thinks there's no mask. This is the last hold up.